### ISSUE TITLE
Unable to edit split transactions

### ISSUE THREAD

**Issue by yurnih** [firefly-iii/firefly-iii#2395]
**Bug description**
During the upgrade from 4.7.17 to 4.8.0, some transactions lost their category (only for grouped revenue/income transactions).

When trying to change the split transaction on the webinterface, the following error is raised;
"Each split must have transaction_journal_id (either valid ID or 0)."

Currently I've changed it in the database itself.

**Steps to reproduce**
1. Edit an existing transaction
2. Add a new split and fill in all required forms (when it doesn't have any split yet)
3. Save the transaction

**Expected behavior**
Change some fields of a transaction without an error.

**Bonus points**
See also demo;
![demo-screenshot](https://user-images.githubusercontent.com/2417500/62832418-d1bebf80-bc2d-11e9-94a3-55c2f6b4adfb.png)

**Comment by JC5** [comment]
Thanks! Not sure what happened there.

**Comment by JC5** [comment]
This will be fixed next release (see the known issues).

**Comment by yurnih** [comment]
Thanks, I've tried it and it works.

Only when I tried it with one of my transactions I found also another issue (somehow related to editing), it is not possible to remove some parts of the split afterwards (except for destroying the whole group and create a new transaction).

In [GroupUpdateService.php#L74](https://github.com/firefly-iii/firefly-iii/blob/develop/app/Services/Internal/Update/GroupUpdateService.php#L74) the actual posted items can differ with the items in the database. 

Should I create a PR for it?
